js/components/portal-dashboard/feedback/rubric-table.tsx

```diff
@@ -0,0 +1,155 @@
+    const ratingId = rating.id;
+    const radioButtonKey = `${critId}-${ratingId}`;
+    const selected = (rubricFeedback && rubricFeedback[critId] && rubricFeedback[critId].id === ratingId);
+    // Tooltips displayed to teacher should actually show student description if it's available.
+    const ratingDescription =
+      (crit.ratingDescriptionsForStudent && crit.ratingDescriptionsForStudent[ratingId]) ||
+      (crit.ratingDescriptions && crit.ratingDescriptions[ratingId]) ||
+      null;
+    const isApplicableRating = crit.nonApplicableRatings === undefined ||
+      crit.nonApplicableRatings.indexOf(ratingId) < 0;
+    const isNotApplicable = !isApplicableRating;
+
+    return (
+      <div className={css.rubricScoreBox} key={radioButtonKey}>
+        <div className={css.rubricButton} title={(isApplicableRating) ? ratingDescription : "Not Applicable"}>
+          { isNotApplicable
+            ? <span className={css.noRating}>N/A</span>
+            : this.renderButton(critId, selected,ratingId, buttonIndex)
+          }
+        </div>
+      </div>
+    );
+  }
+
+  private renderButton = (critId: string, selected: boolean, ratingId: string, buttonIndex: number) => {
+    const handleRatingChange = (buttonIndex: number) => () => {
+      const { rubric, rubricFeedback } = this.props;
+      const deselect = rubric.ratings.findIndex((r: any) => r.id === (rubricFeedback[critId].id)) === buttonIndex;
+
+      updateSelection(critId, ratingId, deselect);
+    };
+
+    const updateSelection = (critId: any, ratingId: string, deselect: boolean) => {
+      const { rubric, rubricFeedback, student } = this.props;
+      const newSelection: any = {};
+      const rating = rubric.ratings.find((r: any) => r.id === ratingId);
+      const criteria = rubric.criteria.find((c: any) => c.id === critId);
+      const score = rating.score;
+      const label = rating.label;
+      const studentId = student.get("id");
+
+      deselect  ? newSelection[critId] = {
+                  id: "",
+                  score: 0,
+                  label: "",
+                  description: "",
+                }
+                : newSelection[critId] = {
+                  id: ratingId,
+                  score,
+                  label,
+                  description: criteria.ratingDescriptions[ratingId],
+                };
+      const newFeedback = Object.assign({}, rubricFeedback, newSelection);
+      this.rubricChange(newFeedback, studentId);
+    };
+
+    return (
+      <button className={css.outerCircle} onClick={handleRatingChange(buttonIndex)}>
+        <div className={`${css.innerCircle} ${selected ? css.selected : ""}`}></div>
+      </button>
+    );
+  }
+
+  private rubricChange = (rubricFeedback: any, studentId: string) => {
+    const { rubric, activityId, activityIndex, updateActivityFeedback } = this.props;
+    let numFeedback = 0;
+    rubric.criteria.map((crit: any)=>{
+      rubricFeedback[crit.id].id !== "" && numFeedback++;
+    });
```

Comment:
I probably would not use a map here because your not mapping each array entry into some new value or output.  You're summing so a forEach might look cleaner.  Something like:
```
    rubric.criteria.forEach((crit: any) => {
      if (rubricFeedback[crit.id].id !== "") {
        numFeedback++
      }
    });
```

Reply:
Similar to the error that @emcelroy mentioned, you might need this as well:
`if (rubricFeedback[crit.id]?.id !== "") {`

Reply:
Basically we need some extra protection for when rubricFeedback doesn't have the entries yet.